0021-hashed-timelock-agreements/0021-hashed-timelock-agreements.md

```diff
@@ -0,0 +1,139 @@
+
+_Internet "hat tip": the importance of being able to integrate **any** type of network was demonstrated by [RFC 1149: A Standard for the Transmission of IP Datagrams on Avian Carriers](https://tools.ietf.org/html/rfc1149)._
+
+## Appendix: Additional HTLA Types
+
+This appendix includes additional HTLA types that are possible but were not mentioned above either because they are less desirable or have yet to be implemented.
+
+### Third Party Escrow
+
+A third party escrow provider can be used if a ledger does not support hashlocks and timeouts but does support fast, inexpensive transfers between accounts. In this case, the sender prepares a transfer by sending the funds to the escrow provider, along with the hashlock and timeout. If the recipient submits the hashlock preimage before the timeout, the escrow provider transfers the funds to the recipient. If not, the escrow provider transfers the funds back.
+
+It is preferable for ledgers to implement holds natively, but this HTLA type may be used in cases where that is not possible or not happening. Both the sender and recipient must trust the escrow provider for the full value of each transfer, either because of reputation or legal arrangement.
+
+### Notarized Payment Channels
+
+Disputes may arise in payment channel-based HTLAs because of disagreements over timeouts. Since there is no single source of truth as to whether the hashlock preimage was submitted in time, clock skew or network delays could cause the recipient to think the transfer should be executed while the sender thinks it expired.
+
+To mitigate the risk of timeout disputes, the sender and recipient could agree to use a third party notary to act as the timekeeper. The sender prepares the transfer by submitting the details to the notary and the recipient (unless the notary also acts as a message broker). The recipient submits the preimage to the notary, who enforces the timeout. The notary signs a statement attesting to whether the preimage was recieved in time and gives the signed message to both the sender and recipient.
+
+Third-party notaries could be used with simple or more complex payment channels. In the simple type, the participants would agree that the notary has the final say about the timeout. In the more complex variety, the payment channels could be constructed such that the channel updates are conditional upon the signed statement from the notary. When using third-party notaries the sender and recipient must trust the notary not to be colluding with one of the parties and to not sign two conflicting statements.
```

Comment:
What if we use a simple payment channel like bitcoin paychan, but the claim signature is the notary's instead of the sender? Or in the ripple case, set the `PublicKey` to the notary's public key instead of the sender.